src/common/commonutils/DeviceInfoUtils.c

```diff
@@ -308,6 +308,27 @@ char* GetCpuModel(void* log)
     return textResult;
 }
 
+unsigned int GetNumberOfCpuCores(void* log)
+{
+    const char* osCpuCoresCommand = "lscpu | grep \"CPU(s):\"";

```

Comment:
There is no need to shell out to call `lscpu`, cpu information is exposed in `/proc/cpuinfo` and the processors can be counted there.

Reply:
Changed to read from /proc/cpuinfo for several that made sense, including this one. It's in last iteration. It makes sense since /proc/cpuinfo is written by Kernel and more direct.